ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -484,7 +493,17 @@ KEXT_STATIC int HandleVnodeOperation(
             if (isRename)
             {
                 // Prevent system services from expanding directories as part of enumeration as this tends to cause deadlocks with the kauth listeners for Antivirus software
-                if (!TryGetVirtualizationRoot(&perfTracer, context, currentVnode, pid, CallbackPolicy_UserInitiatedOnly, &root, &vnodeFsidInode, &kauthResult, kauthError))
+                if (!TryGetVirtualizationRoot(
+                        &perfTracer,
+                        context,
+                        currentVnode,
+                        pid,
+                        CallbackPolicy_UserInitiatedOnly,
+                        false, // allow deleting offline directories even if not expanded
```

Comment:
> `// allow deleting offline directories even if not expanded`

This is inside of `if (isRename)`, and we should not allow renames when the provider is offline.

Should the comment read something like this?:

```
// Until Mojave is set as our minimum version, we must allow offline 
// access when 'isRename' is true as on High Sierra 'isRename' is true for deletes (which should be allowed)
```

Additionally/alternatively should we check `version_major` and only pass in `false` when we're pre-Mojave?

Reply:
I've sorted that out now, with a version-dependent flow.